src/main/java/strategy/BalanceHandlerImpl.java

```diff
@@ -0,0 +1,13 @@
+package strategy;
+
+public class BalanceHandlerImpl implements OperationHandler {
+    private static final int BORDER_POSITIVE = 0;
+
+    @Override
+    public int getAmount(int quantity) {
+        if (quantity < BORDER_POSITIVE) {
+            throw new RuntimeException("Balance can't be negative quantity!");
+        }
+        return quantity;
+    }
```

Comment:
and implementation would be something like
```suggestion
    @Override
    public void processTransaction(FruitTransaction transaction) {
        Storage.getStorage.put(transaction.getFruit(), transaction.getQuantity());
    }
```
Please recheck all handlers